src/facade/dragonfly_connection.cc

```diff
@@ -599,6 +597,16 @@ void Connection::OnPostMigrateThread() {
 void Connection::HandleRequests() {
   ThisFiber::SetName("DflyConnection");
 
+  // We must initialize tl_queue_backpressure_ here and not in the c'tor because a connection object
+  // may be created in a differrent thread from where it runs.
+  if (tl_queue_backpressure_.subscriber_thread_limit == 0) {
```

Comment:
apparently the connection always accessed  tl_queue_backpressure_ from another thread. that may explain why `queue_backpressure_->ec.notifyAll();` was on a profiling radar

Reply:
Good catch 😵

Reply:
had to restructure to support unit tests